connectors/sources/confluence.py

```diff
@@ -256,6 +323,214 @@ async def ping(self):
             logger.exception("Error while connecting to Confluence")
             raise
 
+    async def get_spaces(self):
```

Comment:
I think that should be part of the `ConfluenceClient`

Reply:
`get_spaces` fetches the all the spaces from the confluence. It is similar to `fetch_documents` (fetches pages, blogs) and `fetch_attachments` (files, images, etc.) in terms of functionality, so I believe it should reside in the DataSource class.

I think the naming is a bit misleading. I should've named it as `fetch_spaces` instead. I'll update it.